spring-aop/src/main/java/org/springframework/aop/aspectj/MethodInvocationProceedingJoinPoint.java

```diff
@@ -221,10 +221,11 @@ public Method getMethod() {
 		@Override
 		@Nullable
 		public String[] getParameterNames() {
-			if (this.parameterNames == null) {
-				this.parameterNames = parameterNameDiscoverer.getParameterNames(getMethod());
+			String[] parameterNames = this.parameterNames;
```

Comment:
I really like this. the performance impact is tiny: on `x86` this is "free", on `arm` this would translate to `lsync` (_lightweight_ sync); but it just sets a pattern - which I really like.